pkg/operator/ceph/cluster/controller.go

```diff
@@ -314,20 +314,33 @@ func (c *ClusterController) onUpdate(oldObj, newObj interface{}) {
 	}
 
 	logger.Infof("update event for cluster %s is supported, orchestrating update now", newClust.Namespace)
+
+	// if the image changed, we need to detect the new image version
+	if oldClust.Spec.CephVersion.Image != newClust.Spec.CephVersion.Image {
+		logger.Infof("the ceph version changed. detecting the new image version...")
+		if err := cluster.setCephMajorVersion(15 * time.Minute); err != nil {
+			logger.Errorf("unknown ceph major version. %+v", err)
+			return
+		}
+	} else {
+		logger.Infof("ceph version is still %s on image %s", cluster.Spec.CephVersion.Name, oldClust.Spec.CephVersion.Image)
```

Comment:
we are logging the `Name` field from `cluster` and the `Image` field from `oldClust`? should we log both from `cluster`?

Reply:
yes, they could both be logged from `cluster`

Reply:
are you going to make that change, or is that not worth it?

Reply:
just pushed it as you commented ;)